### DISCUSSION TITLE
How to uninstall googletest

### DISCUSSION THREAD

**Discussion by ljm0451** [google/googletest#3380] [Community Help]
Thanks advance
There is no "make uninstall"

**Comment by adityadees** [comment]
### Uninstall just googletest
`sudo apt-get remove googletest`
_Which will remove just the googletest package itself._

### Uninstall googletest and its dependencies
`sudo apt-get remove --auto-remove googletest`
_This will remove the googletest package and any other dependant packages which are no longer needed._

### Purging your config/data too
_If you also want to delete your local/config files for googletest then this will work._
**_`Caution!   Purged config/data can not be restored by reinstalling the package.`_**

`sudo apt-get purge googletest  `
Or similarly, like this googletest
`sudo apt-get purge --auto-remove googletes`

**Reply by akashblsbrmnm** [reply]
$sudo apt-get remove googletest
[sudo] password for akash: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package 'googletest' is not installed, so not removed
0 upgraded, 0 newly installed, 0 to remove and 12 not upgraded.


It shows googletest is not installed. Any other way ?

**Comment by akashblsbrmnm** [comment]
As of now, i guess "make uninstall" is not yet available. Find `install_manifest.txt` file present inside `build` directory,
The install_manifest.txt lists all the files that were installed. You can manually remove each of these files using the paths listed.
I created a script file to read the install_manifest and removes the files which were uninstalled.
```
#!/bin/bash
while IFS= read -r file
do
  echo "Removing $file"
  sudo rm -f "$file"
done < install_manifest.txt
```

After this, i removed the GTest directory manually from /usr/local/lib/cmake
```
sudo rm -rf GTest
```
Then from /usr/local/include
```
sudo rm -rf gmock/ gtest/
```


This succesfully removed gtest and gmock from my ubuntu machine. Hope the devs add the option for removing the libs from CMakefile :)
